xarray/backends/zarr.py

```diff
@@ -227,6 +233,22 @@ def __getitem__(self, key):
         # if self.ndim == 0:
         # could possibly have a work-around for 0d data here
 
+    async def async_getitem(self, key):
+        array = self._array
+        if isinstance(key, indexing.BasicIndexer):
+            method = self._async_getitem
+        elif isinstance(key, indexing.VectorizedIndexer):
+            # TODO
+            method = self._vindex
+        elif isinstance(key, indexing.OuterIndexer):
+            # TODO
+            method = self._oindex
```

Comment:
@ianhi almost certainly these need to become async to fix your bug

Reply:
This might need a check on the zarr version, else it's going to fail if someone calls this with a version of zarr that isn't recent enough...

Reply:
This has now been resolved, with a fairly complex set of test parametrizations that check for expected behaviour in all the different possible zarr-python versions installed.